### PR TITLE
[fix]: [PL-28241]: add Auth_Settings Edit persmission check on NG LDAP Settings operations

### PR DESCRIPTION
**Pull request by pbarapatre10**
## Describe your changes

Adding permission check for Create/Update/Delete of NG LDAP settings. Only privileged logged in users can update the Auth Settings (LDAP settings)

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37558)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37558)

**Comment by pbarapatre10** [comment]
trigger messagecheck

File: 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java [hunk 1]
```diff
@@ -445,6 +449,8 @@ public class AuthenticationSettingsResource {
   public RestResponse<Boolean>
   deleteLdapSettings(@Parameter(description = ACCOUNT_PARAM_MESSAGE) @QueryParam(
       NGCommonEntityConstants.ACCOUNT_KEY) @NotBlank String accountId) {
+    accessControlClient.checkForAccessOrThrow(
+        ResourceScope.of(accountId, null, null), Resource.of(AUTHSETTING, null), EDIT_AUTHSETTING_PERMISSION);
```
Comment (upadhyay-an) [inline 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java:453]: Should it be `DELETE_AUTHSETTING_PERMISSION`?

Comment (pbarapatre10) [reply 122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java:453]: Yes, will fix this, thanks

**Review by upadhyay-an — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37558#-:-NC-ET2UBUCHNEPKw41B:bnfp4nl)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by kapil-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37558)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by kapil-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37558)* status: 1 file(s) to review. 1. unresolved discussions. 

___
*[`122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java` line 453 at r1](https://reviewable.io/reviews/harness/harness-core/37558#-NBzSORyeyq8oyHRw_A--r1-453:-NC-WBMQDEzS5q47A4Nn:b2ni9ah) ([raw file](https://github.com/harness/harness-core/blob/c7010f8d1a811c3d540cffe25cddbbc774f0d37d/122-ng-authentication-settings/src/main/java/io/harness/ng/authenticationsettings/resources/AuthenticationSettingsResource.java#L453)):*
<details><summary><i>Previously, pbarapatre10 (Prateek Barapatre) wrote…</i></summary><blockquote>

Yes, will fix this, thanks
</blockquote></details>

ok

**Review by ujjawal-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37558#-:-NC-_sV5CEvK2xBt3s1t:bnfp4nl)* status: 1. unresolved discussions.

**Review by upadhyay-an — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37558)* status: :shipit: complete! ✓

**Pull request merged**